Improve guidelines (#926)
commercetools/test-data: docs/guidelines/creating-new-model.md

Since we have two representations for it (REST and GraphQL), we need to export two objects.

> When deciding which model's fields to populate, the rule of thumb is **to only assign values to the required properties**.

Reviewer: ```suggestion
When deciding which model's fields to populate, the rule of thumb is to **only assign values to required properties**.```

Author: Updated here: f4f5bbf9ab378f6b515a97b9e2a087485c49247f

> If you think consumers will most likely need a full-populated version of the model, just create a new preset (`withAllFields`) for it.

Reviewer: ```suggestion
If you expect consumers to need a fully populated version of the model, just create a preset called `withAllFields` for this use case.
```

## Testing

You are expected to implement tests for the builders (REST and GraphQL) that cover the generation of the default version of the data model.
If you have created any preset, please also make sure to write a test file for it validating the properties that the preset populates.

### Validating nested models

If your model has nested models, we need to validate they are correctly populated.

For the REST model we need to use some kind of [Duck typing pattern](https://en.wikipedia.org/wiki/Duck_typing) checking whether the built field has some properties of the expected model.
Example:

```ts
expect(model).toEqual(
  expect.objectContaining({
    id: expect.any(String),
    key: expect.any(String),
    label: expect.objectContaining({
      en: expect.any(String),
      de: expect.any(String),
    }),
  })
);
```

> For the GraphQL version we can rely on the `__typename` property of the built value. If that value is what we expect, we don't need to check for other properties as that property is already indicating the type of the built nested model.

Reviewer: ```suggestion
For the GraphQL version, we can rely on the `__typename` property of the built value. If that value is what we expect, we don't need to check for other properties, as that property is already indicating the type of the built nested model.
```
